qcodes/tests/dataset/test_database_creation_and_upgrading.py

```diff
@@ -48,19 +47,16 @@
 
 @contextmanager
 def location_and_station_set_to(location: int, work_station: int):
-    cfg = Config()
-    old_cfg = deepcopy(cfg.current_config)
+    cfg = qc.config
+    old_cfg = deepcopy(cfg)
     cfg['GUID_components']['location'] = location
     cfg['GUID_components']['work_station'] = work_station
-    cfg.save_to_home()
 
     try:
         yield
 
     finally:
         cfg.current_config = old_cfg
```

Comment:
This is actually wrong. Not a bug introduced here but exposed by rewriting. The `old_cfg` is a `Class Config` but `cfg.current_config` is a `DotDict`